### PR TITLE
refactor campaign: intergrate rewards in user info of campaign + lazy load

### PR DESCRIPTION
**Pull request by nguyenhoaidanh**

### CODE REVIEW

File: src/constants/networks.ts [hunk 1]
```diff
@@ -48,6 +51,7 @@ export const NETWORKS_INFO_CONFIG: NETWORKS_INFO_CONFIG_TYPE = {
   [ChainId.ZKSYNC]: zksync,
   [ChainId.LINEA_TESTNET]: lineaTestnet,
   [ChainId.SOLANA]: solana,
+  [ChainId.SOLANA_TESTNET]: solanaTestnet,
```
Comment (namgold) [inline src/constants/networks.ts:54]: lol

Comment (namgold) [reply src/constants/networks.ts:54]: what would be the usecase

Comment (nguyenhoaidanh) [reply src/constants/networks.ts:54]: campaign claim reward, seperate network.
backend dont need  to update frequently 🙏

File: src/constants/networks.ts [hunk 2]
```diff
@@ -20,13 +20,16 @@ import {
   oasis,
   optimism,
   solana,
+  solanaTestnet,
   velas,
   zksync,
 } from './networks/index'
 import { EVMNetworkInfo } from './networks/type'
 
+type SOLANA_NETWORK = ChainId.SOLANA | ChainId.SOLANA_TESTNET
```
Comment (nhd98z) [inline src/constants/networks.ts:29]: Replace all `TESTNET` to `DEVNET` .

File: src/constants/networks/solana-testnet.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
+import { ChainId } from '@kyberswap/ks-sdk-core'
```
Comment (nhd98z) [inline src/constants/networks/solana-testnet.ts:1]: solana-devnet.ts

File: src/constants/networks/solana-testnet.ts [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    // We might need setup up to 3 accounts or even more for openbook
+    // => use 0.01
+    // above values might change
+    minForGas: 10 ** 7,
+  },
+  aggregatorProgramAddress: 'GmgkeeJtcjHgeiSDdT5gxznUDr5ygq9jo8tmA4ny7ziv',
+  limitOrder: { development: NOT_SUPPORT, production: NOT_SUPPORT },
+  coingeckoNetworkId: 'solana',
+  coingeckoNativeTokenId: 'solana',
+  tokenListUrl: `${KS_SETTING_API}/v1/tokens?chainIds=${ChainId.SOLANA}&isWhitelisted=${true}`,
```
Comment (nhd98z) [inline src/constants/networks/solana-testnet.ts:42]:
```suggestion
  tokenListUrl: `${KS_SETTING_API}/v1/tokens?chainIds=${ChainId.SOLANA_DEVNET}&isWhitelisted=${true}`,
```

File: src/constants/tokens.ts [hunk 5]
```diff
@@ -144,7 +144,7 @@ export const STABLE_COINS_ADDRESS: { [chainId in ChainId]: string[] } = {
   [ChainId.MUMBAI]: [],
   [ChainId.BSCTESTNET]: [],
   [ChainId.AVAXTESTNET]: [],
-  [ChainId.LINEA_TESTNET]: [],
+  [ChainId.SOLANA_TESTNET]: [],
```
Comment (nhd98z) [inline src/constants/tokens.ts:147]: .

File: src/constants/tokens.ts [hunk 6]
```diff
@@ -187,6 +187,7 @@ export const STABLE_COIN_ADDRESSES_TO_TAKE_FEE: Record<ChainId, string[]> = {
   [ChainId.BSCMAINNET]: [],
   [ChainId.ARBITRUM]: [],
   [ChainId.LINEA_TESTNET]: [],
+  [ChainId.SOLANA_TESTNET]: [],
```
Comment (nhd98z) [inline src/constants/tokens.ts:190]: .

File: src/constants/tokens.ts [hunk 7]
```diff
@@ -284,6 +285,7 @@ export const SUPER_STABLE_COINS_ADDRESS: { [chainId in ChainId]: string[] } = {
   [ChainId.BSCTESTNET]: [],
   [ChainId.AVAXTESTNET]: [],
   [ChainId.LINEA_TESTNET]: [],
+  [ChainId.SOLANA_TESTNET]: [],
```
Comment (nhd98z) [inline src/constants/tokens.ts:288]: .

File: src/constants/tokens.ts [hunk 8]
```diff
@@ -332,6 +334,7 @@ export const CORRELATED_COINS_ADDRESS: { [chainId in ChainId]: string[][] } = {
...
```
Comment (nhd98z) [inline src/constants/tokens.ts:337]: .

File: src/constants/tokens.ts [hunk 9]
```diff
@@ -434,6 +437,13 @@ export const DAI: { [chainId in ChainId]: Token } = {
     'DAI',
     'Dai Stablecoin (Wormhole)',
   ),
+  [ChainId.SOLANA_TESTNET]: new Token(
```
Comment (nhd98z) [inline src/constants/tokens.ts:440]: .

File: src/constants/tokens.ts [hunk 10]
```diff
@@ -485,6 +495,13 @@ export const USDC: { [chainId in ChainId]: Token } = {
   [ChainId.OPTIMISM]: new Token(ChainId.OPTIMISM, '0x7F5c764cBc14f9669B88837ca1490cCa17c31607', 6, 'USDC', 'USD Coin'),
   [ChainId.ZKSYNC]: new Token(ChainId.ZKSYNC, '0x3355df6D4c9C3035724Fd0e3914dE96A5a83aaf4', 6, 'USDC', 'USD Coin'),
   [ChainId.SOLANA]: new Token(ChainId.SOLANA, 'EPjFWdd5AufqSSqeM2qN1xzybapC8G4wEGGkZwyTDt1v', 6, 'USDC', 'USD Coin'),
+  [ChainId.SOLANA_TESTNET]: new Token(
```
Comment (nhd98z) [inline src/constants/tokens.ts:498]: .

File: src/constants/tokens.ts [hunk 11]
```diff
@@ -556,6 +573,13 @@ export const USDT: { [chainId in ChainId]: Token } = {
   ),
   [ChainId.ZKSYNC]: new Token(ChainId.ZKSYNC, '0x493257fd37edb34451f62edf8d2a0c418852ba4c', 6, 'USDT', 'Tether USD'),
   [ChainId.SOLANA]: new Token(ChainId.SOLANA, 'Es9vMFrzaCERmJfrF4H2FYD4KCoNkY11McCe8BenwNYB', 6, 'USDT', 'Tether USD'),
+  [ChainId.SOLANA_TESTNET]: new Token(
```
Comment (nhd98z) [inline src/constants/tokens.ts:576]: .

File: src/constants/tokens.ts [hunk 12]
```diff
@@ -655,6 +679,13 @@ export const KNC: { [chainId in ChainId]: Token } = {
     'KNC',
     'Kyber Network Crystal',
   ), // todo namgold: not exists yet
+  [ChainId.SOLANA_TESTNET]: new Token(
```
Comment (nhd98z) [inline src/constants/tokens.ts:682]: .

File: src/index.tsx [hunk 13]
```diff
@@ -22,9 +22,8 @@ import 'swiper/swiper.min.css'
 import SolanaWalletContext from 'components/SolanaWalletContext'
 import { ENV_LEVEL, GTM_ID, MIXPANEL_PROJECT_TOKEN, SENTRY_DNS, TAG } from 'constants/env'
 import { ENV_TYPE } from 'constants/type'
-// import { updateServiceWorker } from 'state/application/actions'
-import CampaignsUpdater from 'state/campaigns/updater'
 
+// import { updateServiceWorker } from 'state/application/actions'
```
Comment (nhd98z) [inline src/index.tsx:26]: delete

File: src/pages/Campaign/CampaignListAndSearch.tsx [hunk 14]
```diff
@@ -33,34 +42,68 @@ const CampaignList = styled.div`
...
+
+  const scroll = useRef<HTMLDivElement | null>(null)
+  const onRefChange = useCallback((node: HTMLDivElement) => {
+    scroll.current = node
+    if (!node?.classList.contains('scrollbar')) {
+      node?.classList.add('scrollbar')
+    }
+  }, [])
+
+  useEffect(() => {
```
Comment (nhd98z) [inline src/pages/Campaign/CampaignListAndSearch.tsx:92]: What is this for?

File: src/pages/Campaign/CampaignListAndSearch.tsx [hunk 15]
```diff
@@ -33,34 +42,68 @@ const CampaignList = styled.div`
...
+    if (scroll.current) scroll.current.scrollTop = 0
+  }, [account, debounceSearch])
+
+  const isItemLoaded = (index: number) => !hasMoreCampaign || index < campaigns.length
+  const itemCount = campaigns.length + Number(hasMoreCampaign)
+
+  function getRowHeight(index: number) {
+    const { showProgressBarNumberTrade, showProgressBarVolume } = getCampaignInfo(campaigns[index], account)
+    const progressbarNum = +showProgressBarNumberTrade + +showProgressBarVolume
+    return campaigns[index]?.status === CampaignStatus.ENDED ? 140 : 160 + progressbarNum * 24
```
Comment (nhd98z) [inline src/pages/Campaign/CampaignListAndSearch.tsx:102]: This look magic, can you add comment to explain?

File: src/state/user/reducer.ts [hunk 16]
```diff
@@ -162,6 +162,7 @@ export const defaultShowLiveCharts: { [chainId in ChainId]: boolean } = {
   [ChainId.BSCTESTNET]: false,
   [ChainId.AVAXTESTNET]: false,
   [ChainId.LINEA_TESTNET]: false,
+  [ChainId.SOLANA_TESTNET]: false,
```
Comment (nhd98z) [inline src/state/user/reducer.ts:165]: .

**Review by nhd98z — CHANGES_REQUESTED** [review]

File: src/pages/Campaign/CampaignListAndSearch.tsx (hunk 14 above)
Comment (nguyenhoaidanh) [reply src/pages/Campaign/CampaignListAndSearch.tsx:92]: scroll list to top when, change search, change network

**Review by nhd98z — APPROVED** [review]

**Pull request merged**